utils.py

```diff
@@ -0,0 +1,142 @@
+import numpy as np
+
+
+def compute_unary(labels, M, GT_PROB=0.5):
+    """
+    Simple classifier that is 50% certain that the annotation is correct
+    (same as in the inference example).
+    """
+    u_energy = -np.log(1.0 / M)
+    n_energy = -np.log((1.0 - GT_PROB) / (M - 1))
+    p_energy = -np.log(GT_PROB)
+
+    U = np.zeros((M, len(labels)), dtype='float32')
+    U[:, labels > 0] = n_energy
+    U[labels, np.arange(U.shape[1])] = p_energy
+    U[:, labels == 0] = u_energy
+    return U
+
+
+def softmax_to_unary(sm, GT_PROB=1):
+    """
+    Util function that converts softmax scores (classwise probabilities) to
+    unary potentials (the negative log likelihood per node).
+
+    Parameters
+    ----------
+    sm: nummpy.array
+        Softmax input. The first dimension is expected to be the classes,
+        all others will be flattend.
+    GT_PROB: float
+        The certainty of the softmax output (default is 1).
+
+    """
+    num_cls = sm.shape[0]
+    if GT_PROB < 1:
+        uniform = np.ones(sm.shape) / num_cls
+        sm = GT_PROB * sm + (1 - GT_PROB) * uniform
+    return -np.log(sm).reshape([num_cls, -1]).astype(np.float32)
+
+
+def create_pairwise_gaussian(sdims, shape):
+    """
+    Util function that create pairwise gaussian potentials. This works for all
+    image dimensions. For the 2D case does the same as
+    `DenseCRF2D.addPairwiseGaussian`.
+
+    Parameters
+    ----------
+    sdims: list or tuple
+        The scaling factors per dimension. This is referred to `sxy` in
+        `DenseCRF2D.addPairwiseGaussian`.
+    shape: list or tuple
+        The shape the CRF has.
+
+    """
+    # create mesh
+    cord_range = [range(s) for s in shape]
+    mesh = np.array(np.meshgrid(*cord_range, indexing='ij'), dtype=np.float32)
+
+    # scale mesh accordingly
+    for i, s in enumerate(sdims):
+        mesh[i] = mesh[i] / s
```

Comment:
I don't see why `mesh[i] /= s` should not work. I will test and update it in the PR.